docs/source/common_issues.rst

```diff
@@ -51,6 +51,11 @@ flagged as an error.
 
   If you don't know what types to add, you can use ``Any``, but beware:
 
+- **The function returns ``None`` value but it was annotated as having a return
```

Comment:
I would re-word this to something like "A function annotated as non-Optional returns None and mypy doesn't complain".